stl/inc/experimental/random

```diff
@@ -0,0 +1,55 @@
+// random experimental header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _EXPERIMENTAL_RANDOM_
+#define _EXPERIMENTAL_RANDOM_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#include <random>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+namespace experimental {
+    inline namespace fundamentals_v3 {
+
+        inline default_random_engine& _Engine() {
+            thread_local default_random_engine _Eng{random_device{}()};
+            return _Eng;
+        }
+
+        template <class _IntType>

```

Comment:
I think it would be better to SFINAE that out. Otherwise that function would still show up in IDEs.

Reply:
In addition to being an observable difference, the error message for a vanished SFINAE overload is considerably worse (for C1XX) than the error message for a `static_assert`, so we should do what the ~~Standard~~ TS says here.